trino/client.py

```diff
@@ -913,15 +927,27 @@ def _get_time_with_timezome(self, value, time_size, pattern):
             tz = timedelta(hours=int(matches.group(3)), minutes=int(matches.group(4)))
         return datetime.strptime(matches.group(1)[:time_size], pattern).time().replace(tzinfo=timezone(tz))
 
+    def _get_time_with_timezone_trunc_ms(self, value, time_size, ms_div, pattern):
+        matches = re.match(r'^(.*)([\+\-])(\d{2}):(\d{2})$', value)
+        assert matches is not None
+        assert len(matches.groups()) == 4
+        if matches.group(2) == '-':
+            tz = -timedelta(hours=int(matches.group(3)), minutes=int(matches.group(4)))
+        else:
+            tz = timedelta(hours=int(matches.group(3)), minutes=int(matches.group(4)))
+        time_str = matches.group(1)[:time_size]
+        ms_str = str(round(int(matches.group(1)[time_size:]) / ms_div))
+        return datetime.strptime(time_str + ms_str, pattern).time().replace(tzinfo=timezone(tz))
+
     def _get_number_of_digits(self, column):
```

Comment:
this method is doing too many things and it's unclear what the return value means and method name doesn't help either

Reply:
Fixed